local-storage/src/block.rs

```diff
@@ -0,0 +1,263 @@
+use anyhow::{bail, Result};
+use bytes::Bytes;
+use cid::Cid;
+use iroh_unixfs::Block;
+use std::collections::HashSet;
+use std::str::FromStr;
+
+#[derive(Debug, PartialEq)]
+pub struct StoredBlock {
+    pub cid: String,
```

Comment:
You might consider replacing these strings with string slices.

Reply:
Hmm I would love to do this, but there are two places where I create a `StoredBlock` using temp data that falls out of scope (1. from the results from a db query, 2. in a test harness from random generated bytes). I'll dig into this more tomorrow but I'm not sure it's worth the refactoring (right now) to get that optimization.